spectrocloud/resource_virtual_machine.go

```diff
@@ -0,0 +1,368 @@
+package spectrocloud
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/spectrocloud/hapi/models"
+	"github.com/spectrocloud/palette-sdk-go/client"
+	"github.com/spectrocloud/terraform-provider-spectrocloud/spectrocloud/schemas"
+	"strings"
+)
+
+func resourceVirtualMachine() *schema.Resource {
```

Comment:
@nikchern can we please ensure every single attribute gets a description, please?

Reply:
@karl-cardenas-coding sure, plan is to add description before merge PR. There will be one small round of review with you tomorrow.